### PR TITLE
Make the linter Dart 2 clean

### PR DESCRIPTION
**Pull request by bwilkerson**

### CODE REVIEW

File: lib/src/util/leak_detector_visitor.dart [hunk 1]
```diff
@@ -71,8 +71,9 @@ _VisitVariableDeclaration _buildVariableReporter(
 
 Iterable<AstNode> _findMethodCallbackNodes(Iterable<AstNode> containerNodes,
     VariableDeclaration variable, Map<DartTypePredicate, String> predicates) {
-  Iterable<PrefixedIdentifier> prefixedIdentifiers =
-      containerNodes.where((n) => n is PrefixedIdentifier);
+  Iterable<PrefixedIdentifier> prefixedIdentifiers = containerNodes
+      .where((n) => n is PrefixedIdentifier)
+      .cast<PrefixedIdentifier>();
```
Comment (a14n) [inline lib/src/util/leak_detector_visitor.dart]: `containerNodes.whereType<PrefixedIdentifier>()`

File: lib/src/util/leak_detector_visitor.dart [hunk 2]
```diff
@@ -83,8 +84,9 @@ Iterable<AstNode> _findMethodCallbackNodes(Iterable<AstNode> containerNodes,
 
 Iterable<AstNode> _findMethodInvocationsWithVariableAsArgument(
     Iterable<AstNode> containerNodes, VariableDeclaration variable) {
-  Iterable<MethodInvocation> prefixedIdentifiers =
-      containerNodes.where((n) => n is MethodInvocation);
+  Iterable<MethodInvocation> prefixedIdentifiers = containerNodes
+      .where((n) => n is MethodInvocation)
+      .cast<MethodInvocation>();
```
Comment (a14n) [inline lib/src/util/leak_detector_visitor.dart]: `containerNodes.whereType<MethodInvocation>()`

File: lib/src/util/leak_detector_visitor.dart (hunk 1 above)
Comment (bwilkerson) [reply lib/src/util/leak_detector_visitor.dart]: done

File: lib/src/util/leak_detector_visitor.dart (hunk 2 above)
Comment (bwilkerson) [reply lib/src/util/leak_detector_visitor.dart]: done

File: test/mocks.dart [hunk 3]
```diff
@@ -2,14 +2,17 @@
...
 import 'dart:io';
 
 import 'package:analyzer/analyzer.dart';
 import 'package:analyzer/src/generated/engine.dart';
 import 'package:analyzer/src/generated/source.dart';
 import 'package:analyzer/src/lint/linter.dart';
+import 'package:analyzer/src/lint/project.dart';
 import 'package:analyzer/src/lint/pub.dart';
-import 'package:mockito/mockito.dart';
+import 'package:front_end/src/scanner/token.dart';
```
Comment (pq) [inline test/mocks.dart]: This dependency is unfortunate.  Hoping this is just a short term thing?

File: test/formatter_test.dart [hunk 4]
```diff
@@ -3,9 +3,9 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:analyzer/error/error.dart';
+import 'package:analyzer/src/generated/engine.dart';
 import 'package:analyzer/src/lint/linter.dart';
 import 'package:linter/src/formatter.dart';
-import 'package:mockito/mockito.dart';
```
Comment (pq) [inline test/formatter_test.dart]: 👍

**Comment by bwilkerson** [comment]
I have no idea why the builds have failed. Yesterday when I ran the tests locally, both with and without the `--preview-dart-2` flag, everything passed. Today there are failures either way when I run it locally. I'll have to dig into it, unless someone has an idea.

**Comment by a14n** [comment]
I can see the same error locally.

**Comment by a14n** [comment]
Here's the underlying cause:

```
UnimplementedError: whereType is not yet supported
#0      _HashFieldBase&_HashBase&_OperatorEqualsAndHashCode&SetMixin.whereType (dart:collection/set.dart:68)
#1      _findMethodCallbackNodes (package:linter/src/util/leak_detector_visitor.dart:75:22)
```

I hadn't tested `whereType` before. Looks like it's not ready yet. Is it? @lrhn

**Comment by a14n** [comment]
See dart-lang/sdk#32463 . Looks like you have to revert the last commit to avoid `whereType`.

**Comment by lrhn** [comment]
It doesn't work in Dart 1 mode (where there are no type arguments to generic methods).
That made a lot of people scared when people started using the method anyway, developing in DDC and deploying with dart2js. So, it's disabled until we disable Dart 1. Which is hopefully Real Soon Now.

**Comment by a14n** [comment]
Regarding https://github.com/dart-lang/sdk/issues/32463#issuecomment-371771585 the code with `cast` may also have problems.

**Comment by a14n** [comment]
@lrhn thanks!

**Comment by bwilkerson** [comment]
Is there an easy way to roll back on GitHub? My machine died, so I lost my local commit history.

**Comment by a14n** [comment]
@bwilkerson I created another PR with your 2 first commits :#953

**Pull request closed without merging**